base/range.jl

```diff
@@ -585,9 +585,10 @@ function show(io::IO, r::LinRange{T}) where {T}
     print(io, "LinRange{")
     show(io, T)
     print(io, "}(")
-    show(io, first(r))
+    ioc = IOContext(io, :typeinto=>T)
+    show(ioc, first(r))
```

Comment:
This still seems backwards, BTW. Why print `LinRange{Float32}(1.0, 2.0, 3)` rather than `LinRange(1.0f0, 2.0f0, 3)`? Notice how other ranges print:
```
julia> range(1,2f0,3) |> println
1.0f0:0.5f0:2.0f0

julia> range(1,2f0,3) |> typeof
StepRangeLen{Float32, Float64, Float64, Int64}

julia> LinRange(1,2f0,3) |> println
LinRange{Float32}(1.0, 2.0, 3)

julia> LinRange(1,2f0,3) |> typeof
LinRange{Float32, Int64}
```

Reply:
Because this makes it more clear what the eltype is. Ranges printed as `a:b:c` have no way to display their eltype at the collection level so it makes sense to do it implicitly at the element level, but in general (e.g. Array, Set, Dict, Base.Pairs) it is standard to print eltype at the collection level.

Reply:
Like this you mean?
```
julia> Dict(1=>2) |> println
Dict(1 => 2)

julia> Dict(1=>2) |> typeof
Dict{Int64, Int64}

julia> Set([1,2,3]) |> println
Set([2, 3, 1])

julia> Set([1,2,3]) |> typeof
Set{Int64}
```
They show more detail in display, but in compact printing, prefer to be compact.

Reply:
I don't know the rule is quite what you state. These want to be compact & hide the type, although their display versions would not:
```
julia> Dict(1=>2) |> println
Dict(1 => 2)

julia> Set([1,2,3]) |> println
Set([2, 3, 1])
```

Collections show it on the grounds, I assume, that showing it once is usually more compact than showing it N times:
```
julia> Dict([i=>i/2f0 for i in 1:10]) |> println
Dict{Int64, Float32}(5 => 2.5, 4 => 2.0, 6 => 3.0, 7 => 3.5, 2 => 1.0, 10 => 5.0, 9 => 4.5, 8 => 4.0, 3 => 1.5, 1 => 0.5)

julia> collect(1:10f0) |> println
Float32[1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]

julia> Set([1f0,2,3]) |> println  # still avoids Set{Float32} as redundant
Set(Float32[2.0, 3.0, 1.0])
```
But LinRange is not printed like a collection.

```
julia> Pair(1, 2f0) |> println
1 => 2.0f0
```